inference/trillium/JetStream-Maxtext/DeepSeek-R1-671B/README.md

```diff
@@ -0,0 +1,533 @@
+export CLUSTER_CKPT_NODE_DISK_SIZE=3000
+export CLUSTER_CKPT_NUM_NODES=1
+export ARTIFACT_REGISTRY_REPO_NAME=jetstream-maxtext-ar
+export ARTIFACT_REGISTRY=${REGION}-docker.pkg.dev/${PROJECT_ID}/${ARTIFACT_REGISTRY_REPO_NAME}
+export JETSTREAM_MAXTEXT_IMAGE=jetstream-maxtext
+export JETSTREAM_MAXTEXT_VERSION=latest
+export HF_MODEL_NAME="deepseek-ai/DeepSeek-R1"
+export MODEL_NAME=deepseek3-671b
+export GCS_CKPT_PATH_BF16=gs://${GCS_BUCKET}/models/${MODEL_NAME}/bf16
+export GCS_CKPT_PATH_UNSCANNED=gs://${GCS_BUCKET}/models/${MODEL_NAME}/unscanned
+```
+
+Following are required variables that must be set:
+
+- `<PROJECT_ID>`: your Google Cloud project ID  
+- `<REGION>`: the region where you want to run Cloud Build  
+- `<CLUSTER_NAME>`: the name of your GKE cluster  
+- `<CLUSTER_ZONE>`: the zone where your cluster is located  
+- `<GCS_BUCKET>`: the name of your Cloud Storage bucket. Do not include the gs:// prefix  
+- `<TPU_RESERVATION>`: the name of the TPU reservation
+
+Following are required variables with default values already set:
+
+- `TPU_TYPE`: TPU accelerator type supported by TPU v6e. Refer to the [supported list](https://cloud.google.com/tpu/docs/v6e#configurations).   
+- `NUM_SLICES`: The number of slices to use  
+- `CLUSTER_CPU_MACHINE_TYPE`: The CPU nodepool machine type  
+- `CLUSTER_CKPT_NODEPOOL_NAME`: The name of CPU nodepool used for checkpoint conversion  
+- `CLUSTER_CKPT_NODE_MACHINE_TYPE`: The machine type of CPU nodepool used for checkpoint conversion  
+- `CLUSTER_CKPT_NODE_DISK_SIZE`: The disk size of CPU nodepool used for checkpoint conversion. For this recipe, a minimum of 3TB disk size is suggested.  
+- CLUSTER\_CKPT\_NUM\_NODES  
+- `ARTIFACT_REGISTRY`: the full name of your Artifact Registry in the following format: *LOCATION*\-docker.pkg.dev/*PROJECT\_ID*/*REPOSITORY*  
+- `JETSTREAM_MAXTEXT_IMAGE`: the name of the JetStream MaxText image  
+- `JETSTREAM_MAXTEXT_VERSION`: the version of the JetStream MaxText image
+
+Set the default project:
+
+``` bash
+gcloud config set project $PROJECT_ID
+```
+
+## Create GKE Cluster with TPU v6e nodepool using xpk
+
+Use a custom network for better performance as well as to avoid having the default network becoming overloaded. Please refer to the [network performance optimizations](https://cloud.google.com/tpu/docs/v6e-intro/#network_performance_optimizations) for more details.
+
+``` bash
+export NETWORK_NAME_1=${CLUSTER_NAME}-mtu9k-1
+export NETWORK_FW_NAME_1=${NETWORK_NAME_1}-fw-1
+
+# Use a custom network for better performance as well as avoid the default network to be overloaded.
+gcloud compute networks create ${NETWORK_NAME_1} --mtu=8896 --project=${PROJECT_ID} --subnet-mode=auto --bgp-routing-mode=regional
+gcloud compute firewall-rules create ${NETWORK_FW_NAME_1} --network ${NETWORK_NAME_1} --allow tcp,icmp,udp --project=${PROJECT_ID}
+
+# Secondary subnet for multinic experience. Need custom ip routing to be different from first network’s subnet.
```

Comment:
The secondary network seems to be only needed for [multislice / multi-nic setups](https://cloud.google.com/tpu/docs/v6e-intro/#network_performance_optimizations:~:text=The%20following%20environment%20variables%20are%20needed%20for%20a%20secondary%20subnet%20when%20you%20are%20using%20a%20Multislice%20environment). However the cluster created here uses `NUM_SLICES=1`. So this can be skipped?